src/Catalyst.Modules.Server.Blazor/AutofacServiceProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace Catalyst.Modules.Server.Blazor
+{
+    /// <summary>
+    /// A factory for creating a <see cref="ContainerBuilder"/> and an <see cref="IServiceProvider" />.
+    /// </summary>
+    public class AutofacServiceProviderFactory : IServiceProviderFactory<ContainerBuilder>
+    {
+        private readonly ContainerBuilder _containerBuilder;
+        private AutofacServiceProvider _autofacServiceProvider;
+
+        public AutofacServiceProviderFactory(ContainerBuilder containerBuilder)
+        {
+            _containerBuilder = containerBuilder;
+        }
+
+        /// <summary>
+        /// Creates a container builder from an <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The collection of services.</param>
+        /// <returns>A container builder that can be used to create an <see cref="IServiceProvider" />.</returns>
+        public ContainerBuilder CreateBuilder(IServiceCollection services)
+        {
+            _containerBuilder.Populate(services);
+            return _containerBuilder;
+        }
+
+        public void SetContainer(IContainer container) { _autofacServiceProvider.LifetimeScope = container; }
+
+        /// <summary>
+        /// Creates an <see cref="IServiceProvider" /> from the container builder.
+        /// </summary>
+        /// <param name="containerBuilder">The container builder.</param>
+        /// <returns>An <see cref="IServiceProvider" />.</returns>
+        public IServiceProvider CreateServiceProvider(ContainerBuilder containerBuilder)
+        {
+            if (containerBuilder == null) throw new ArgumentNullException(nameof(containerBuilder));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)